rust/experimental/lang/src/statestack.rs

```diff
@@ -101,3 +97,43 @@ impl<T: Clone + Hash + Eq, N: NewState<T>> Context<T, N> {
         result
     }
 }
+
+pub struct HolderNewState<T> {
```

Comment:
ditto my earlier doc comment... comment, a bit if info here would save me having to actually understand the _code_ 😉